### PR TITLE
WIP[Plugin Extensions]: Panel Chrome UX

### PR DESCRIPTION
**Pull request by jackw**
**What is this feature?**

This adds a separate extensions menu to the Panel Chrome header inline with [the figma design](https://www.figma.com/file/kgRjDmubFvlGg2r0oTmhFU/UI-exstensions?node-id=8-119&t=VhIn4ZYlody5L0ax-0).

**Why do we need this feature?**

To provide a better UX for plugin extensions that register link at the `grafana/dashboard/panel/menu` extension point. I've opted to use the recently added actions prop to add this to panel chrome. I've wrapped PanelMenu for Panel Chrome specific menus otherwise I was having to duplicate some styles and components. This felt like a cleaner option.

_What does it look like?_

https://user-images.githubusercontent.com/73201/230128053-756154e8-acc9-4aac-a3f9-cdf2b63e1efb.mp4

**Who is this feature for?**

Grafana users

**Which issue(s) does this PR fix?**:

Related #61658

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard/utils/getPanelHeaderExtensionsMenu.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { PanelModel, DashboardModel } from '../state';
+
+import { truncateTitle } from './getPanelMenu';
+
+interface Props {
+  panel: PanelModel;
+  dashboard: DashboardModel;
+}
+
+export function getPanelHeaderExtensionsMenu({ panel, dashboard }: Props): PanelMenuItem[] {
```
Comment (mckn) [inline public/app/features/dashboard/utils/getPanelHeaderExtensionsMenu.ts:14]: So one question is if we should try to call this `getPanelHeaderExtensionPoint`

**Review by mckn — APPROVED** [review]
LGTM!

File: public/app/features/dashboard/utils/getPanelHeaderExtensionsMenu.ts (hunk 1 above)
Comment (mckn) [reply public/app/features/dashboard/utils/getPanelHeaderExtensionsMenu.ts:14]: Also - I'm not sure I think that is is a good idea to group the items below their plugin. Would an alternative way be to put the plugin name below the item as a description? Similar to how this component work. https://developers.grafana.com/ui/latest/index.html?path=/story/forms-select--select-with-option-descriptions

@Ijin08

File: public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx [hunk 2]
```diff
@@ -693,6 +696,19 @@ export class PanelStateWrapper extends PureComponent<Props, State> {
         </div>
       );
 
+      const extensionMenuItems = getPanelHeaderExtensionsMenu({ panel, dashboard });
+      let actions = [];
+      if (extensionMenuItems.length > 0) {
+        actions.push(
+          <PanelChromeMenu
```
Comment (torkelo) [inline public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx:703]: I don't think we want more than one menu per panel, that seems really messy

File: public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx [hunk 3]
```diff
@@ -693,6 +696,19 @@ export class PanelStateWrapper extends PureComponent<Props, State> {
         </div>
       );
 
+      const extensionMenuItems = getPanelHeaderExtensionsMenu({ panel, dashboard });
```
Comment (torkelo) [inline public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx:699]: I don't want to load all plugins just to render a dashboard, can we do this lazy when the menu os opened?

**Review by torkelo — COMMENTED** [review]
Is there a doc with uses cases and designs a L.M. additional panel menu?

**Comment by jackw** [comment]
Closing as after discussing with UX design is considered PoC and shouldn't be introduced to Grafana.

**Pull request closed without merging**